search-service-bing-geocoder/src/main/java/nl/aerius/search/tasks/BingSearchService.java

```diff
@@ -76,29 +94,58 @@ public Single<SearchTaskResult> retrieveSearchResults(final String query) {
   }
 
   private void retrieveSuggestions(final String query, final Map<String, SearchSuggestion> sugs) {
-    final String url = String.format(PDOK_SUGGEST_ENDPOINT, query, apiKey);
+    final String url = String.format(BING_SUGGEST_ENDPOINT, query, apiKey);
 
+    // First obtain suggestions, then translate them to actual locations.
     final HttpResponse<JsonNode> json = Unirest.get(url).asJson();
     final JSONObject body = json.getBody().getObject();
 
-    final JSONArray arr = body.getJSONArray("resourceSets").getJSONObject(0).getJSONArray("resources");
+    final JSONArray arr = body
+        .getJSONArray("resourceSets").getJSONObject(0)
+        .getJSONArray("resources").getJSONObject(0)
+        .getJSONArray("value");
+    // As the list of suggestions can contain duplicates when we translate them to actual locations, filter those out by using a set
+    // Use a LinkedHashSet to keep the order.
+    final Set<SuggestedLocation> suggestedLocations = new LinkedHashSet<>();
     for (int i = 0; i < arr.length(); i++) {
-      final JSONObject jsonObject = arr.getJSONObject(i);
-
-      // Skip if the result is not in GB
-      if (!jsonObject.getJSONObject("address").getString("countryRegionIso2").equals(REGION)) {
-        continue;
```

Comment:
If I recall correctly then the bing search service could return results outside of GB, even when specifying the search region, which is why this additional filter was introduced -- does the suggest service also do this?

Reply:
(For example `York` would also return the pennsylvanian city)

Reply:
did try `york`, and got `United Kingdom` back as 2nd result.... at least it's all in GB, but it can use some more improvements ;)